pilot/pkg/xds/monitoring.go

```diff
@@ -167,6 +167,12 @@ var (
 		monitoring.WithLabels(typeTag),
 	)
 
+	proxySDSCertificateErrors = monitoring.NewGauge(
+		"proxy_sds_certificate_errors",
+		"Number of failed to fetch key and certificate.",
+		monitoring.WithLabels(nodeTag),
```

Comment:
Please do not use unbounded value as metric label.

Reply:
I saw https://github.com/istio/istio/blob/3b9c6b88e0696577f4f1d0e93d7c28912828790d/pilot/pkg/xds/monitoring.go#L59 is also used like this.

Is there any suggestions?
Use secret name and ns as label?

Reply:
Looks like secret name and namespace has already be logged. IMO that should be good enough, and this metric does not necessarily need to have any label. Alert can be built with this metric and user could check istiod logs to troubleshoot.

Also metric name probably should be `pilot_sds_certificate_errors_total` for consistency with exisiting metric as well as prometheus recommendation: https://prometheus.io/docs/practices/naming/

Reply:
thanks very much!